Ensure JSON content type in performRequestOnCurrentHost

benlabbe opened the pull request.

benlabbe committed:
Ensure JSON content type in performRequestOnCurrentHost

benlabbe committed:
Configure a cmake package to deploy Elasticlient
This includes the export of the externals (cpr, jsoncpp)

benlabbe committed:
update readme

mbumba committed:
Adding of JSON content type header to HTTP requests which having body.

mbumba committed:
Merge remote-tracking branch 'github/json-content-type' into master

mbumba commented:
Hello @benlabbe, thank you for your participation on Elasticlient library. At first, I would like to say sorry for my late reply. Few minutes ago I merged the fix (https://github.com/seznam/elasticlient/commit/5d91a855f058c61257ad0b4034e380c674cde2e7) which should ensure that JSON content type is sent when request's body is not empty. I also modified tests for this issue.

Does this solve the issue for you? This pull request goes from your master branch, so I'm not entirely sure which commits are relevant. Thank you

WLBF committed:
add updateDocument to SameIndexBulkData

mbumba committed:
Merge pull request #5 from WLBF/feature/bulk-update

updateDocument feature for SameIndexBulkData

kwach committed:
Provide a way to connect via proxy

Proxy can by used to 
 - debug connection
 - change headers on the flight when needed

kwach committed:
Add separate constructor to maintain ABI

mbumba committed:
Update include/elasticlient/client.h

Co-Authored-By: kwach <marek@kwasecki.com>

mbumba committed:
Update include/elasticlient/client.h

Co-Authored-By: kwach <marek@kwasecki.com>

mbumba committed:
Update src/client-impl.h

Co-Authored-By: kwach <marek@kwasecki.com>

mbumba committed:
Update src/client.cc

Co-Authored-By: kwach <marek@kwasecki.com>

kwach committed:
Fix online merge fail

mbumba committed:
Merge pull request #7 from radpointhq/add-proxy-capability

Providing a way to connect to ES cluster via proxy server

johniez committed:
#8: Do not move temporary - moving a temporary object prevents copy elision.

johniez committed:
Merge branch 'copy-elision'

mbumba committed:
Fixes issue #13

johniez committed:
#12: Externals updated to latest versions.

johniez committed:
Merge pull request #15 from seznam/externals-update

Externals updated to latest versions.

johniez committed:
Elasticlient ssl options by new header-only constructor.

johniez committed:
Add possibility to use password for the ssl key file.

johniez committed:
Readme for options arguments.

johniez committed:
Doxygen mainpage - added reference to new options.

mbumba committed:
Retained compatibility with older ES versions and improved tests for Scroll.

Signed-off-by: Martin Bumba <mbumba@email.cz>

mbumba committed:
Merge pull request #17 from seznam/issue-13-unsupported-scroll-syntax

Fixes issue #13

johniez committed:
Merge pull request #16 from ssl-options

johniez committed:
Status code type was changed in https://github.com/whoshuu/cpr/commit/350b0f81e6ab70b47e0eef7089b68da07ff491e9, format it as long...

ladislavmacoun committed:
Changed document identifier to be optional

While creating a document with an explicit identifier, Elasticsearch
needs to perform a check whether the same id already exists within the
same shard.

This patch makes setting document id optional. And using
auto-generated ids by leaving _id parameter, which makes indexing faster.

Signed-off-by: Ladislav Macoun <ladislavmacoun@gmail.com>

ladislavmacoun committed:
Updated doxy strings

Signed-off-by: Ladislav Macoun <ladislavmacoun@gmail.com>

johniez committed:
Merge pull request #20 from ladislavmacoun/make-doc-id-optional

Changed document identifier to be optional

ladislavmacoun committed:
Added option to turn off document validation

For long string the validation is expensive, and often not necessary
since the doc could be validated during its
constructions/deserialization.

The option default to true, so previous API should not be broken.

Signed-off-by: Ladislav Macoun <ladislavmacoun@gmail.com>

ladislavmacoun committed:
Add option build elasticlient lib statically

The library will be build depending on the BUILD_SHARED_LIBS[1] flag.
This enables to build and link this library statically.

[1]: https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

Signed-off-by: Ladislav Macoun <ladislavmacoun@gmail.com>

johniez committed:
Bump of httpmockserver deps.

johniez committed:
Merge remote-tracking branch 'origin/master' into static-build

johniez committed:
Remove BUILD_STATIC_LIBS variable, only BUILD_SHARED_LIBS is valid choice.
Added BUILD_SHARED_LIBS to readme.

johniez committed:
#21: Passing compiler flag -DELASTICLIENT_STATICLIB is not necessary.
So versions are set only when building shared library.

johniez committed:
Merge branch 'static-build'

johniez committed:
#22: Keep ABI compatibility by not changing mangled symbols by adding optional argument. When ABI breaks this commit could be "reverted"...

johniez committed:
Merge branch 'disable-doc-validation'

johniez committed:
Github Workflow cmake action for master branch.

johniez committed:
Replace Travis doc build with GH Actions.

benlabbe committed:
update FindCPR.cmake

benlabbe committed:
Merge remote-tracking branch 'seznam/master' into merging_from_seznam

# Conflicts:
#	src/CMakeLists.txt
#	src/client.cc

benlabbe committed:
Better handeling of system  libraries while producing config.cmake file

benlabbe committed:
Merge branch 'master' of github.com:benlabbe/elasticlient

# Conflicts:
#	src/CMakeLists.txt
#	src/client.cc

benlabbe committed:
Merge branch 'merging_from_seznam'

# Conflicts:
#	src/client.cc

benlabbe commented:
Dear Martin,

I recently merged all your contributions since 2018 into my master branch. Still my pull request is relevant from my point of view because I updated the cmake config scripts for an easier package distribution / integration in other C++ projects .

Any remarks ?
